### PR TITLE
Extend the checks to make sure only real user admins can create the accounts

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #22603.

### Summary of Changes

As discussed #22603 lets add core.manage to the required permissions

#### Requirements
- enable admin activation
- create an admin group that is not allowed to core.manage com_users
- create an real admin (default group)
- create an admin from the new group
- both users with sendMail enabled

#### Normal case
- register an account
- make sure only the account with core.manage gets the mail
- make sure the Registration completes

#### Non-normal case
- register an account
- open the link from the mail
- but login as user who does not have core.mange in com_users
- make sure there is a error message.

### Expected result

add core.mange to the required permissions 

### Actual result

core.manage is not required to create accounts from the Frontend.

### CODE REVIEW

File: components/com_users/controllers/registration.php [hunk 1]
```diff
@@ -75,7 +75,7 @@ public function activate()
 		if (($uParams->get('useractivation') == 2) && $userToActivate->getParam('activate', 0))
 		{
 			// If a user admin is not logged in, redirect them to the login page with an error message
-			if (!$user->authorise('core.create', 'com_users'))
+			if (!$user->authorise('core.create', 'com_users') && !$user->authorise('core.manage', 'com_users'))
```
Comment (bembelimen) [inline components/com_users/controllers/registration.php]: Shouldn't this be a ```||``` instead of ```&&```?

**Comment by HLeithner** [comment]
Doesn't looks like a good solution.

File: components/com_users/controllers/registration.php (hunk 1 above)
Comment (zero-24) [reply components/com_users/controllers/registration.php]: It is `&&` following the logic that you need both permissions to create the user from the backend.

**Comment by zero-24** [comment]
> Doesn't looks like a good solution.

Any suggestions? And why does this not look like a good solution? It makes sure only users that can create users from the backend can also create / approve them from the Frontend.

**Comment by ReLater** [comment]
Shall we test or not?

**Comment by zero-24** [comment]
> Shall we test or not?

Lets wait for @HLeithner 's answer on my question first.

**Comment by HLeithner** [comment]
if you don't have a better solution we can merge it but should really find a better solution.

**Comment by zero-24** [comment]
> if you don't have a better solution we can merge it but should really find a better solution.

Well can you explain what is your issue with the current solution? Maybe we can find a better solution than ;)

**Comment by zero-24** [comment]
any update @HLeithner ?

**Comment by ghost** [comment]
@HLeithner gently reminder.

**Comment by HLeithner** [comment]
As we don't have a better solution at the moment I would merged this if we get some tests.

**Comment by zero-24** [comment]
Great thanks, than happy testing 👍

File: components/com_users/controllers/registration.php (hunk 1 above)
Comment (SharkyKZ) [reply components/com_users/controllers/registration.php]: But you need only 1 permission here?

Comment (zero-24) [reply components/com_users/controllers/registration.php]: @SharkyKZ Sorry i missed your comment. Why should you only need one permission here. That is the point of adding the check here.

You need both as also normal authors have by default core.create permissions.

**Comment by zero-24** [comment]
@joomla/bug-squad @joomla/security would be great to get some tests here as this also has a sec impact. :)

File: components/com_users/controllers/registration.php (hunk 1 above)
Comment (SharkyKZ) [reply components/com_users/controllers/registration.php]: In that case it does have to be `||` here.

Comment (zero-24) [reply components/com_users/controllers/registration.php]: Sorry got the point now, fixed. Thanks

**Comment by infograf768** [comment]
Drone is not happy. Des not look related to the patch.

**Comment by richard67** [comment]
@infograf768 Drone shows merge conflicts in the "clone" step ... so maybe that **is** related to this PR.

**Comment by zero-24** [comment]
Drone / clone is happy now.

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on 79cc673c58309dd5924a369415408a435de23f50<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24738">issues.joomla.org/tracker/joomla-cms/24738</a>.</sub>

**Comment by conseilgouz** [comment]
I have tested this item :white_check_mark: successfully on 79cc673c58309dd5924a369415408a435de23f50<br /><br />Just had problems to understand which parameters to disable to fit test requirements....<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24738">issues.joomla.org/tracker/joomla-cms/24738</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24738">issues.joomla.org/tracker/joomla-cms/24738</a>.</sub>

**Comment by zero-24** [comment]
Thanks for the tests :+1:

**Comment by HLeithner** [comment]
Thanks

**Comment by zero-24** [comment]
Thanks

**Comment by KrejtSparck** [comment]
Well, this change broke the possibility to allow for a more or less regular user to activate new user accounts from the front end. I don't think that was intended? Did you take that into account with the change you made? and did you test that?

Two days ago I updated to Joomla 3.9.19. Suddenly I notice my colleagues (who have no Joomla knowledge at all!) can no longer activate new user accounts through the activation link (https://xxxxxx.nl/log-in?task=registrat ... 82xxxxxxxx) provided in the System Email sent to them.
(The "New User Account Activation" setting is set to 'Administrator'.) Trying to activate a user gives an error message: "U bent niet gemachtigd om nieuwe accounts te activeren, meld u aan met een bevoorrecht account". Something like "You are not authorized to activate new accounts. Provide a privileged account".

The user account used to activate the new users from the frontend is a member of the User Groups 'Registered' and 'Publisher'. With this user account and these memberships I could always activate new users. Notice this user account is not (and was never) a member of the User Groups 'Manager' or 'Administrator'.

Adding the user account used to activate the new users to the User Group 'Administrator' fixed the problem. With that membership I can activate new users. But I don't want this account to have the rights associated with the 'Administrator' group! (Like access to the backend administration panel.)

I think this consequence was overseen???

**Comment by SharkyKZ** [comment]
It was not overseen. It was the intention of this PR. The idea is that only "administrators" can activate users. Although you don't have to use Administrator user group. You can create a custom group with the right permissions for `com_users`.

**Comment by KrejtSparck** [comment]
Yeah, you can,... I know quite a lot about using Joomla and I maintain several Joomla sites, but fiddling with user groups and rights is a whole different cookie, That really is too difficult and risky for Joe Average Joomla Admin! Please consider to go back to the drawing board?

**Comment by infograf768** [comment]
It is far more risky to let Registered and Publisher default groups activate new users.

**Comment by KrejtSparck** [comment]
@HLeithner You were hesitant about this change. What do you think about my point? This change indeed makes sure only users that can create users from the backend can also create/approve them from the Frontend, but it blocks users that have no admin skills/rights to approve new users from the frontend.

**Comment by KrejtSparck** [comment]
@infograf768 and making them admin is less risky?
com_users
Anyhow, these users with Registered and Publisher rights do get create, edit and edit_state on top of edit_own on com_users by default. Instead of && !$user->authorise('core.manage', 'com_users') removing these permissions would have been sufficient?

**Comment by SharkyKZ** [comment]
Yes, realistically, low privileged users should not have create permissions in com_users.

**Comment by HLeithner** [comment]
@KrejtSparck I see no problem with this PR, the concern was that you we don't have the possibility to combine rights in joomla, somehting like you need "right x" to have "right y". The requirement is not wrong if you can't manage user you shouldn't be able to manage them.

**Pull request merged**